src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -139,8 +133,12 @@ def _execute(self, all_targets):
   class InvalidShardSpecification(TaskError):
     """Indicates an invalid `--test-shard` option."""
 
+  def _ensure_workdir(self):
```

Comment:
If that change passes CI, I'd be fine with it.

But the recommended caching/workdir strategy these days is to use either `Task.cache_target_dirs` or `Task.create_target_dirs` to automatically have per-target directories created in the workdir. So manually creating a workdir should be getting rarer.